### PR TITLE
Add likes and dislikes functional

### PR DESCRIPTION
**Pull request by MiBBy-4**

### CODE REVIEW

File: app/controllers/reactions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  def index
+    current_customer_reactions = @current_customer.reactions
+    render json: {
+      current_customer_reactions: current_customer_reactions,
+    }
+  end
+
+  def create
+    @reaction = @current_customer.reactions.new(reaction_params)
+    if !@reaction.save
```
Comment (AndreyAnischik) [inline app/controllers/reactions_controller.rb:14]: make the first condition positive `if @reaction.save`, also send the reaction back

File: app/controllers/reactions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    else
+      render json: {
+        status: :ok,
+      }
+    end
+  end
+
+  def destroy
+    @reaction = @current_customer.reactions.find(params[:id])
+    idea = @reaction.idea
```
Comment (AndreyAnischik) [inline app/controllers/reactions_controller.rb:27]: do you need this `idea`?

File: db/migrate/20220222102859_add_fields_to_customer_and_idea_tables.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class AddFieldsToCustomerAndIdeaTables < ActiveRecord::Migration[6.1]
+  def change
+    add_column :customers, :name, :string
+    add_column :customers, :surname, :string
+    add_column :customers, :phone_number, :string
+    add_column :customers, :skype, :string
+    add_column :ideas, :avg_likes, :float
```
Comment (AndreyAnischik) [inline db/migrate/20220222102859_add_fields_to_customer_and_idea_tables.rb:7]: for what do we need these fields?

**Review by AndreyAnischik — CHANGES_REQUESTED** [review]

File: db/migrate/20220222102859_add_fields_to_customer_and_idea_tables.rb (hunk 3 above)
Comment (MiBBy-4) [reply db/migrate/20220222102859_add_fields_to_customer_and_idea_tables.rb:7]: They are need for businessmen that answer on an idea. So, it was in my technical task.

File: db/migrate/20220222135354_add_boolean_filed_in_reactions_table.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddBooleanFiledInReactionsTable < ActiveRecord::Migration[6.1]
+  def change
+    add_column :reactions, :reaction, :bool
```
Comment (AndreyAnischik) [inline db/migrate/20220222135354_add_boolean_filed_in_reactions_table.rb:3]: rename the column please. is is strange that in reactions table there a reaction field.

File: db/migrate/20220222102859_add_fields_to_customer_and_idea_tables.rb (hunk 3 above)
Comment (AndreyAnischik) [reply db/migrate/20220222102859_add_fields_to_customer_and_idea_tables.rb:7]: can't you just count them when there is a need?

File: app/controllers/reactions_controller.rb [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    current_customer_reactions = @current_customer.reactions
+    render json: {
+      current_customer_reactions: current_customer_reactions,
+    }
+  end
+
+  def create
+    @reaction = @current_customer.reactions.new(reaction_params)
+    idea_id = @reaction.idea_id
+    if Reaction.exists?(customer_id: session[:customer_id], idea_id: idea_id)
```
Comment (AndreyAnischik) [inline app/controllers/reactions_controller.rb:15]: use `current_customer[:id]`

File: app/controllers/reactions_controller.rb [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      exist_reaction = Reaction.where(customer_id: session[:customer_id], idea_id: idea_id).last
+      if exist_reaction.reaction === nil
```
Comment (AndreyAnischik) [inline app/controllers/reactions_controller.rb:17]: exist_reaction.reaction.nil?

File: app/controllers/reactions_controller.rb [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        exist_reaction.attributes = { reaction: @reaction.reaction }
+        exist_reaction.save
+        render json: {
+          status: 'Change from nil to bool',
```
Comment (AndreyAnischik) [inline app/controllers/reactions_controller.rb:21]: return existing http status, but also an instance of reaction

File: app/controllers/reactions_controller.rb [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        }
+      else
+        if (exist_reaction.reaction === true && @reaction.reaction === false) || (exist_reaction.reaction === false && @reaction.reaction === true)
```
Comment (AndreyAnischik) [inline app/controllers/reactions_controller.rb:24]: `exist_reaction.reaction? && !@reaction.reaction?` ?

File: app/controllers/reactions_controller.rb [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+          exist_reaction.attributes = { reaction: @reaction.reaction }
+          exist_reaction.save
+          render json: {
+            status: 'Bool reaction was changed'
```
Comment (AndreyAnischik) [inline app/controllers/reactions_controller.rb:28]: add instance, return existing http status

File: app/controllers/reactions_controller.rb [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+          }
+        else
+          exist_reaction.attributes = { reaction: nil }
+          exist_reaction.save
+          render json: {
+            status: 'Change to nil'
```
Comment (AndreyAnischik) [inline app/controllers/reactions_controller.rb:34]: add instance, return existing http status

File: app/controllers/reactions_controller.rb [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+        end
+      end
+    else
+      if @reaction.save
+        render json: {
+          status: :ok,
+        }
+      else
+        render json: {
+          status: 500,
```
Comment (AndreyAnischik) [inline app/controllers/reactions_controller.rb:45]: why 500?

File: app/controllers/reactions_controller.rb [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+          exist_reaction.save
+          render json: {
+            status: 'Change to nil'
+          }
+        end
+      end
+    else
+      if @reaction.save
+        render json: {
+          status: :ok,
```
Comment (AndreyAnischik) [inline app/controllers/reactions_controller.rb:41]: add instance

**Review by AndreyAnischik — CHANGES_REQUESTED** [review]

File: app/controllers/reactions_controller.rb [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+      end
+    else
+      if @reaction.save
+        render json: {
+          status: 200,
+          reaction: @reaction,
+        }
+      else
+        render json: {
+          status: 400,
```
Comment (AndreyAnischik) [inline app/controllers/reactions_controller.rb:48]: why is it 400?

**Review by AndreyAnischik — CHANGES_REQUESTED** [review]

**Review by AndreyAnischik — APPROVED** [review]

**Pull request merged**